helm/cloudflared/templates/_clean+delete_argo_tunnel.tpl

```diff
@@ -0,0 +1,54 @@
+{{- define "clean+delete-argo-tunnel-command" -}}
+command:
+  - sh
+  - -c
+  - |
+    #!/bin/bash
+
+    set -o errexit
+    set -o pipefail
+    set -o nounset
+
+    ACCOUNT_ID="${ACCOUNT_ID:-}"
+    API_KEY="${API_KEY:-}"
+    TUNNEL_NAME="NotSetYet"
+    TUNNEL_ID="NotSetYet"
+
+    API_ENDPOINT="https://api.cloudflare.com/client/v4/accounts/${ACCOUNT_ID}/tunnels"
+
+    set_tunnel_id() {
+      curl -s -X GET \
+        -H "Authorization: Bearer ${API_KEY}" \
+        -H "Content-Type: application/json" \
+        ${API_ENDPOINT}\?name=${TUNNEL_NAME}\&is_deleted=false \
+        -o /tmp/getTunnel.json
+
+      if jq -e '.[] | select(.id != null)' /tmp/getTunnel.json &>/dev/null
+      then
+        TUNNEL_ID=$(jq -r '.[].id' /tmp/getTunnel.json)
+        echo "Found tunnel, using Tunnel ID ${TUNNEL_ID}"
+      else
+        echo "ERR: No Tunnel : ${TUNNEL_NAME}"
+        exit 1
+      fi
+    }
+
+    clean_delete_tunnel() {
+      curl -s -X DELETE \
+        -H "Authorization: Bearer ${API_KEY}" \
+        -H "Content-Type: application/json" \
+        ${API_ENDPOINT}/${TUNNEL_ID}/connections \
+        -o /tmp/cleanTunnel.json
+
+      curl -s -X DELETE \
+        -H "Authorization: Bearer ${API_KEY}" \
+        -H "Content-Type: application/json" \
+        ${API_ENDPOINT}/${TUNNEL_ID}
+    }
+
+    {{ range until 2 }}
```

Comment:
Could this loop be written in bash? The `2` smells of magic number

Reply:
Yeh, I considered doing the loop in bash but I wanted to keep the bash script as simple as possible.. It was kind of a trade off.

`2` kind of is magic and is bound by the recommendation of creating Argo Tunnels in Kubernetes.  Technically you can >=2, it's related to the number of tunnels created/tunning. But I stuck with 2 as that is what the docs recommend.

It would have been nice if I could use like a constant, but Helm doesn't support such things :(

However, I've just had a thought about using a named template... I'll have a play with that